Client/Application/Shortcuts/ActivityShortcutManager.swift

```diff
@@ -0,0 +1,203 @@
+        switch self {
+            case .newTab:
+                return Strings.Shortcuts.activityTypeNewTabSuggestedPhrase
+            case .newPrivateTab:
+                return Strings.Shortcuts.activityTypeNewPrivateTabSuggestedPhrase
+            case .clearBrowsingHistory:
+                return Strings.Shortcuts.activityTypeClearHistorySuggestedPhrase
+            case .enableBraveVPN:
+                return Strings.Shortcuts.activityTypeEnableVPNSuggestedPhrase
+            case .openBraveToday:
+                return Strings.Shortcuts.activityTypeOpenBraveTodaySuggestedPhrase
+            case .openPlayList:
+                return Strings.Shortcuts.activityTypeOpenPlaylistSuggestedPhrase
+        }
+    }
+}
+
+/// Singleton Manager handles creation and action for Activities
+class ActivityShortcutManager: NSObject {
+    
+    /// Custom Intent Types
+    enum IntentType {
+        case openWebsite
+        case openHistory
+        case openBookmarks
+    }
+
+    // MARK: Lifecycle
+    
+    static var shared = ActivityShortcutManager()
+    
+    // MARK: Activity Creation Methods
+    
+    public func createShortcutActivity(type: ActivityType) -> NSUserActivity {
+        let attributes = CSSearchableItemAttributeSet(itemContentType: kUTTypeItem as String)
+        attributes.contentDescription = type.description
+        
+        let activity = NSUserActivity(activityType: type.identifier)
+        activity.persistentIdentifier = NSUserActivityPersistentIdentifier(type.identifier)
+        
+        activity.isEligibleForSearch = true
+        activity.isEligibleForPrediction = true
+        
+        activity.title = type.title
+        activity.suggestedInvocationPhrase = type.suggestedPhrase
+        activity.contentAttributeSet = attributes
+
+        return activity
+    }
+
+    // MARK: Activity Action Methods
+
+    public func performShortcutActivity(type: ActivityType, using bvc: BrowserViewController) {
+        switch type {
+            case .newTab:
+                bvc.openBlankNewTab(attemptLocationFieldFocus: true, isPrivate: false)
+            case .newPrivateTab:
+                bvc.openBlankNewTab(attemptLocationFieldFocus: true, isPrivate: true)
+            case .clearBrowsingHistory:
+                History.deleteAll {
+                    bvc.tabManager.clearTabHistory() {
+                        bvc.openBlankNewTab(attemptLocationFieldFocus: true, isPrivate: false)
+                    }
+                }
+            case .enableBraveVPN:
+                bvc.openBlankNewTab(attemptLocationFieldFocus: true, isPrivate: false)
+
+                switch BraveVPN.vpnState {
+                    case .notPurchased, .purchased, .expired:
+                        guard let vc = BraveVPN.vpnState.enableVPNDestinationVC else { return }
+                    
+                        let settingsNavigationController = SettingsNavigationController(rootViewController: vc)
```

Comment:
Same here, feels like placing this logic in bvc makes more sense, i know our bvc code is growing bigger and bigger :D
I do not think this manager class should handle all these details

Reply:
Moved Navigational functions into Browser. e1e25ad